src/controllers/user.ts

```diff
@@ -0,0 +1,95 @@
+// Assuming you have the UserModel defined and imported from user.model.ts
+import { Request, Response } from 'express'
+import UserModel, { IUser } from '../models/User' // Make sure to import the IUser interface
+
+// Helper function for basic email validation
+const isValidEmail = (email: string): boolean => {
+  const emailRegex = /^[\w-]+(\.[\w-]+)*@([\w-]+\.)+[a-zA-Z]{2,7}$/
+  return emailRegex.test(email)
+}
+
+// Create a new user
+export const createUser = async (req: Request, res: Response): Promise<Response> => {
```

Comment:
```suggestion
export const createUser = async (req: Request, res: Response, next: NextFunction): Promise<Response | void> => {

```
use next to handle erro in async instead of try-catch